osquery/filesystem/filesystem.cpp

```diff
@@ -444,8 +444,25 @@ inline Status listInAbsoluteDirectory(const fs::path& path,
 Status listFilesInDirectory(const fs::path& path,
                             std::vector<std::string>& results,
                             bool recursive) {
-  return listInAbsoluteDirectory(
-      (path / ((recursive) ? "**" : "*")), results, GLOB_FILES);
+  if (path.empty() || !pathExists(path)) {
+    return Status(1, "Target directory is invalid");
+  }
+
+  if (recursive) {
```

Comment:
I don't know these libraries very well -- if there's a symlink loop here, what happens?

Reply:
If there is I believe that risks already exists as a result of this PR... https://github.com/osquery/osquery/pull/7974#issuecomment-1540613721

I'm reusing the code that was introduced for directory listing to list files here. So I'm also uncertain as to whether there is a potential symlink loop... but if there is I believe it'll already exist 👀